src/pages/DuetHome.tsx

```diff
@@ -54,6 +58,18 @@ const DuetHome: React.FC = () => {
     }
   };
 
+  const handleRoomIdPaste = (e: React.ClipboardEvent<HTMLDivElement>) => {
+    e.preventDefault();
+    const roomLinkRegex = /^.*\?id=(?<roomId>\d{0,4})$/;
```

Comment:
hm... why would you try to paste a link into the pin tho, if its just the pin then it makes sense, but the whole link seems funny

Reply:
Part of the reason is that we need the copy link functionality no matter we allow them to copy only the ID or not, and I don't know how to structure the UI so that they can copy both. 🤣

Reply:
Another reason is that in case they copied the link instead of the ID (regardless of whether we give them a convenient button to copy the ID separately), pasting the link and realising that they copied the wrong thing is kinda annoying. Since we already have the information we need, might as well fill in the room id.